lib/SILOptimizer/Utils/CanonicalOSSALifetime.cpp

```diff
@@ -295,13 +296,16 @@ endsAccessOverlappingPrunedBoundary(SILInstruction *inst) {
 //
 // If the lifetime extension did not iterate, then def would be destroyed within
 // B's access scope when originally it was destroyed outside that scope.
-void CanonicalizeOSSALifetime::extendLivenessThroughOverlappingAccess() {
-  this->accessBlocks = accessBlockAnalysis->get(getCurrentDef()->getFunction());
+void CanonicalOSSALivenessInfo::extendLivenessThroughOverlappingAccess(DominanceInfo *domTree,
+                                                                       NonLocalAccessBlocks *accessBlocks) {
+  if (consumingBlocks.empty())
+    return;
 
   // Visit each original consuming use or destroy as the starting point for a
   // backward CFG traversal. This traversal must only visit blocks within the
   // original extended lifetime.
   bool changed = true;
+
   while (changed) {
     changed = false;
     blockWorklist.initializeRange(consumingBlocks);
```

Comment:
@eeckstein @atrick This is the place where I ran into weird memory behavior like weirdness behavior when having a BasicBlockWorklist.

The changes I made were:

1. I put in the check above that we hit the loop with at least one consuming block so we always have at least one block that we can use to initialize BasicBlockWorklist. (I subsequently kept that check if empty since I think it is a nice invariant and with time I think we will want to use BasicBlockWorklist in some form here.
2. I changed initializeRange/insert to pushIfNotVisited.

Shouldn't that just work?